controllers/monitoring_alert.go

```diff
@@ -167,6 +176,38 @@ func (r *ManagedMCGReconciler) reconcileNoobaaRules() error {
 			return fmt.Errorf("failed to unmarshal noobaa-rules.yaml: %w", err)
 		}
 
+		// addLabel is a helper function to add a particular label to all rules in the PrometheusRuleSpec.
+		addLabel := func(label, value string) {
```

Comment:
I think that using the `external_labels` setting in prometheus configuration would fit very well with our requirements, (and less code is always better.) In any case your proposal is ok  for me.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/ 
```
# The labels to add to any time series or alerts when communicating with
# external systems (federation, remote storage, Alertmanager).
external_labels:
    [ [<labelname>]: [<labelvalue>]
```

Reply:
Tracking this here: https://github.com/red-hat-storage/mcg-osd-deployer/issues/72. Will come back to it and replace this approach if the `external_labels` thing fulfills our use case once I have a bit more b/w in the future.